OpenShift/03_create_cluster.sh

```diff
@@ -53,6 +53,37 @@ function extract_installer() {
     export OPENSHIFT_INSTALLER="${outdir}/openshift-baremetal-install"
 }
 
+function rhcos_image_url() {
+
+#Dont do anything if there is a value already set
+if [[ -z "${RHCOS_IMAGE_URL}" ]]
+then
+  # Get the git commit that the openshift installer was built from
+  OPENSHIFT_INSTALL_COMMIT=$($OPENSHIFT_INSTALLER version | grep commit | cut -d' ' -f4)
+
+  # Get the rhcos.json for that commit
+  OPENSHIFT_INSTALLER_RHCOS=${OPENSHIFT_INSTALLER_RHCOS:-https://raw.githubusercontent.com/openshift/installer/$OPENSHIFT_INSTALL_COMMIT/data/data/rhcos.json}
+
+  # Get the rhcos.json for that commit, and find the baseURI and openstack image path
+  RHCOS_IMAGE_JSON=$(curl "${OPENSHIFT_INSTALLER_RHCOS}")
+  RHCOS_INSTALLER_IMAGE_URL=$(echo "${RHCOS_IMAGE_JSON}" | jq -r '.baseURI + .images.openstack.path')
+  export RHCOS_IMAGE_URL=${RHCOS_IMAGE_URL:-${RHCOS_INSTALLER_IMAGE_URL}}
+
+fi
+}
+get_provision_if() {
+
+#Dont do anything if there is a value already set
+if [[ -z "${PRO_IF}" ]]
```

Comment:
Looks great, just one comment: can we use here a variable name different than PRO_IF? 

PRO_IF is already used in 02_configure_host.sh and for the vm env we need to pass a different value to gen_metal3_config.sh than in 02_configure_host.sh

Reply:
Actually I think that won't be needed because we run make prep in a different step.

Reply:
It seems that we do need a different variable name for the virt envs because PRO_IF is already defined in cloud_${USER}.sh which gets loaded in common.sh

Reply:
I tested with and with out specifying PRO-IF and it deployed the  for me.


I will make the 2 changes suggested above:
PRO_IF - will go with INTERNAL_NIC
Will change the cache.